src/extension/features/toolkit-reports/pages/balance-over-time/BalanceOverTime.jsx

```diff
@@ -53,7 +73,31 @@ export const BalanceOverTimeComponent = ({ allReportableTransactions, filters })
       }
     });
 
-    if (shouldGroupAccounts) {
+    if (shouldGroupAccounts && shouldGroupAccountsByType) {
+      // Group datapoints by account type
+      let groupedData = {
+        entities: {},
+        ids: [],
+      };
+      filteredData.forEach((datapoints, accountId) => {
+        const { accountType } = getEntityManager().getAccountById(accountId);
+
+        if (!groupedData.entities[accountType]) {
+          groupedData.entities[accountType] = [];
+          groupedData.ids.push(accountType);
+        }
+
+        groupedData.entities[accountType].push(datapoints);
+      });
+
+      groupedData.ids.forEach(accountType => {
+        newSeries.push({
+          name: GROUPED_BY_TYPE_LABELS[accountType],
```

Comment:
should just be able to use the `l10n` helper here and remove your constant

```javascript
l10n(`accountTypes.${accountType}`)
```

Reply:
Done